public/app/features/alerting/unified/components/expressions/Expression.tsx

```diff
@@ -0,0 +1,328 @@
+          {!editingRefId && (
+            <div className={styles.editable} onClick={() => setEditMode('refId')}>
+              <div className={styles.expression.refId}>{refId}</div>
+            </div>
+          )}
+          {editingRefId && (
+            <AutoSizeInput
+              autoFocus
+              defaultValue={refId}
+              minWidth={5}
+              onChange={(event) => {
+                onUpdateRefId(event.currentTarget.value);
+                setEditMode(false);
+              }}
+              onFocus={(event) => event.target.select()}
+              onBlur={(event) => {
+                onUpdateRefId(event.currentTarget.value);
+                setEditMode(false);
+              }}
+            />
+          )}
+          {!editingType && (
+            <div className={styles.editable} onClick={() => setEditMode('expressionType')}>
+              <div className={styles.mutedText}>{capitalize(queryType)}</div>
+              <Icon size="xs" name="pen" className={styles.mutedIcon} onClick={() => setEditMode('expressionType')} />
+            </div>
+          )}
+          {editingType && (
+            <Select
+              isOpen
+              autoFocus
+              onChange={(selection) => {
+                onUpdateExpressionType(selection.value ?? ExpressionQueryType.classic);
+                setEditMode(false);
+              }}
+              onBlur={() => {
+                setEditMode(false);
+              }}
+              options={gelTypes}
+              value={selectedExpressionType}
+              width={25}
+            />
+          )}
+        </Stack>
+        <Spacer />
+        <IconButton name="trash-alt" variant="secondary" className={styles.mutedIcon} onClick={onRemoveExpression} />
+      </Stack>
+    </header>
+  );
+};
+
+interface FrameProps extends Pick<ExpressionProps, 'isAlertCondition'> {
+  frame: DataFrame;
+}
+
+const FrameRow: FC<FrameProps> = ({ frame, isAlertCondition }) => {
+  const styles = useStyles2(getStyles);
+
+  const name = getSeriesName(frame);
+  const value = getSeriesValue(frame);
+
+  const showFiring = isAlertCondition && value !== 0;
+  const showNormal = isAlertCondition && value === 0;
+
+  return (
+    <div className={styles.expression.resultsRow}>
+      <Stack direction="row" gap={1} alignItems="center">
+        <span className={cx(styles.mutedText, styles.expression.resultLabel)} title={name}>
+          {name}
+        </span>
+        {/* TODO format the values with american notation or SI notation? */}
+        <div className={styles.expression.resultValue}>{value}</div>
+        {showFiring && <AlertStateTag state={PromAlertingRuleState.Firing} size="sm" />}
+        {showNormal && <AlertStateTag state={PromAlertingRuleState.Inactive} size="sm" />}
+      </Stack>
+    </div>
+  );
+};
+
+// TODO: move this to design system or util file?
+const Spacer = () => <span style={{ flex: 1 }}></span>;
```

Comment:
💬 Any thought on whether we should move it somewhere? It's currently just a `div` I've used to push flex elements away from each other.

Reply:
`justify-content: space-between` behaves a bit differently, it's essentially the same of only two child components are involved and the `<Flex />` component is in the center, but if the `<Flex />` component is in another place (such as `<div /><div /><Flex />`) it's not going to work with just `justify-content` :(